docs/guides/sqlite3_db_with_concord.md

```diff
@@ -0,0 +1,259 @@
+
+The first step in this guide will be how to open the DB file, as it's required for the other sqlite3 functions like read and write data.
+
+```c
+#include <sqlite3.h>
+
+...
+
+sqlite3 *db;
+int rc = sqlite3_open("db.sqlite", &db);
+
+if (rc != SQLITE_OK) {
+    log_fatal("[SQLITE] Error when opening the db file. [%s]", sqlite3_errmsg(db));
+      
+    if (sqlite3_close(db) != SQLITE_OK) {
+        log_fatal("[SQLITE] Failed to close sqlite db. [%s]", sqlite3_errmsg(db));
+        abort();
+    }
+    return;
+}
+```
+
+In the first line, we are simply including the sqlite3 into the file.
+
+Then, we are **opening the DB file** with `sqlite3_open` and name it db.sqlite (NOTE: If the file doesn't exist, sqlite3 will create it for you). After opening, it will **check if the status of the `sqlite3_open` is not equal to `SQLITE_OK`** (that represents that it was successfully executed), it's not equal, it will log a fatal error about it, getting the error message with `sqlite3_errmsg` and then closing the DB with `sqlite3_close`. (NOTE: **Yes, even failing to open, you MUST use `sqlite3_close` as said in sqlite3 docs**)
+
+> Whether or not an error occurs when it is opened, resources associated with the database connection handle should be released by passing it to sqlite3_close() when it is no longer required.
+
+Rarely this happens, but if the `sqlite3_close` **fails to be executed**, we will be just killing the bot's process. Since this is not a highly detailed guide, we won't be teaching how to deal with it with many details, but you will only need to finish the finalize the ongoing statement and then use `sqlite3_close` again.
+
+### Creating a sqlite3 table
+
+Firstly, before trying to save data into the DB file, we will need to create a sqlite3 table first. sqlite3 tables contain a schema for that DB. The schema for a DB is "a description of all of the other tables, indexes, triggers, and views that are contained within the database".
+
+```c
+char *msgErr = NULL;
+char *query = sqlite3_mprintf("CREATE TABLE IF NOT EXISTS concord_guides(user_id INT, guild_id INT);");
+rc = sqlite3_exec(db, query, NULL, NULL, &msgErr);
+
+sqlite3_free(query);
+
+if (rc != SQLITE_OK) {
+    log_fatal("[SQLITE] Something went wrong while creating concord_guides table. [%s]", msgErr);
+    if (sqlite3_close(db) != SQLITE_OK) {
+        log_fatal("[SQLITE] Failed to close sqlite db. [%s]", sqlite3_errmsg(db));
+        abort();
+    }
+    return;
+}
+
+sqlite3_close(db);
+```
+
+Here, we are defining the command that **will be executed** with `sqlite3_mprintf`, the command going to be executed is "`CREATE TABLE IF NOT EXISTS concord_guides(user_id INT, guild_id INT)`", where **if concord_guides table doesn't exist**, it is going to be created.
+
+Also, we are saving **INT values**, but if you wanted to **save a text TEXT value**, you can simply replace the `INT` with `TEXT`. (WARN: **This will cause you to do some modifications when retrieving data**)
```

Comment:
"Also, we are saving **INT values**, but if you wanted to **save a text value (like a string)**, you can simply replace `INT` with `TEXT` (WARNING: **This will cause you to have to do some modifications when retrieving data**)."

Reply:
Thanks, fixed